contracts/SwapAndBridge.sol

```diff
@@ -0,0 +1,150 @@
+    /// This affects first swap of every token by leaving 1 wei on the contract.
+    /// @param executor Aggregation executor that executes calls described in `data`
+    /// @param desc Swap description
+    /// @param permit Should contain valid permit that can be used in `IERC20Permit.permit` calls.
+    /// @param data Encoded calls that `caller` should execute in between of swaps
+    /// @return returnAmount Resulting token amount
+    /// @return spentAmount Source token amount     
+    */
+    function swap(
+        IAggregationExecutor executor,
+        SwapDescription calldata desc,
+        bytes calldata permit,
+        bytes calldata data
+    ) external payable returns (uint256 returnAmount, uint256 spentAmount);
+}
+
+/**
+ * @title SwapAndBridge
+ * @notice Allows caller to swap tokens on a chain and bridge them via Across atomically.
+ */
+contract SwapAndBridge is Lockable, MultiCaller {
+    using SafeERC20 for IERC20;
+
+    USSSpokePoolInterface public immutable spokePool;
+    I1InchAggregationRouterV5 public immutable oneInchRouter;
+
+    event SwapAndBridge1Inch(
+        I1InchAggregationRouterV5.SwapDescription swapDescription,
+        IAggregationExecutor aggregationExecutor
+    );
+
+    constructor() {
+        // Addresses are hard-coded below for Optimism-network for demonstration purposes. In production, these
+        // would be passed in as constructor args. They are left here to help debugging and testing.
+        spokePool = USSSpokePoolInterface(0x6f26Bf09B1C792e3228e5467807a900A503c0281);
+        oneInchRouter = I1InchAggregationRouterV5(0x11111112542D85B3EF69AE05771c2dCCff4fAa26);
+    }
+
+    /**
+     * @notice Swaps tokens on this chain via 1Inch and bridges them via Across atomically. Caller can fully specify
+     * their slippage tolerance for the swap and also the full Across deposit params.
+     * @param aggregationExecutor Address of 1inch contract that executes calls described in `oneInchData`.
+     * @param swapDescription 1Inch SwapDescription struct, packed to consolidate function params:
+     *     - srcToken: Token to pull from msg.sender into contract and swap on 1inch into dstToken.
+     *     - dstToken: Token to receive from 1inch swap and to be deposited into Across.
+     *     - srcReceiver: Address to receive `srcToken` from msg.sender. Overwritten by this function to be this contract.
+     *     - dstReceiver: Address to receive `dstToken` from 1inch swap. Overwritten by this function to be this contract.
+     *     - amount: Amount of `srcToken` to swap. Return amount is deposited into Across.
+     *     - minReturnAmount: Minimum amount of `dstToken` to receive from 1inch swap.
+     *     - flags: Overwritten by this contract.
+     * @param outputToken Token to receive from Across deposit on destination chain.
+     * @param outputAmount of `outputToken` to receive from Across deposit on
+     * destination chain.
+     */
+    function swap1InchAndBridge(
```

Comment:
Do you think we should pass in the min amount we expect to get back from 1inch separately so we can check? Or do you think we should fully rely on their contract?

Reply:
yep, +1 d5ba1dae353361f71de766a2acb63cae8b7bd2c9